monai/networks/blocks/fft_utils_t.py

```diff
@@ -0,0 +1,229 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, Sequence
+
+import torch
+from torch import Tensor
+
+from monai.utils.type_conversion import convert_data_type
+
+
+def roll_1d(x: Tensor, shift: int, shift_dim: int) -> Tensor:
+    """
+    Similar to roll but for only one dim.
+
+    Args:
+        x: input data (k-space or image) that can be
+            1) real-valued: the shape is (C,H,W) for 2D spatial inputs and (C,H,W,D) for 3D, or
+            2) complex-valued: the shape is (C,H,W,2) for 2D spatial data and (C,H,W,D,2) for 3D. C is the number of channels.
+        shift: the amount of shift along each of shift_dims dimension
+        shift_dim: the dimension over which the shift is applied
+
+    Returns:
+        1d-shifted version of x
+
+    Note:
+        This function is called when fftshift and ifftshift are not available in the running pytorch version
+    """
+    shift = shift % x.size(shift_dim)
+    if shift == 0:
+        return x
+
+    left = x.narrow(shift_dim, 0, x.size(shift_dim) - shift)
+    right = x.narrow(shift_dim, x.size(shift_dim) - shift, shift)
+
+    return torch.cat((right, left), dim=shift_dim)
+
+
+def roll(x: Tensor, shift: Sequence[int], shift_dims: Sequence[int]) -> Tensor:
+    """
+    Similar to np.roll but applies to PyTorch Tensors
+
+    Args:
+        x: input data (k-space or image) that can be
+            1) real-valued: the shape is (C,H,W) for 2D spatial inputs and (C,H,W,D) for 3D, or
+            2) complex-valued: the shape is (C,H,W,2) for 2D spatial data and (C,H,W,D,2) for 3D. C is the number of channels.
+        shift: the amount of shift along each of shift_dims dimensions
+        shift_dims: dimensions over which the shift is applied
+
+    Returns:
+        shifted version of x
+
+    Note:
+        This function is called when fftshift and ifftshift are not available in the running pytorch version
+    """
+    assert len(shift) == len(shift_dims)
```

Comment:
https://deepsource.io/gh/Project-MONAI/MONAI/run/e294f502-6d56-4658-960f-6ba4a789994c/python/

Reply:
Oh I forgot to check. @mersad95zd , `assert` should not be used except for in the unit test. You may do an `if` `else` check and `raise Error` if not pass.

Reply:
I've updated the PR, and also include a patch to unblock merging (https://github.com/Project-MONAI/MONAI/issues/4606)